mne/io/meas_info.py

```diff
@@ -1211,3 +1214,141 @@ def _empty_info():
+    lpa : array, shape (1, 3)
+        The position of the left periauricular fidicual point in
+        the RAS head space.
+    rpa : array, shape (1, 3)
+        The position of the right periauricular fidicual point in
+        the RAS head space.
+    """
+    def __init__(self, hsp, hpi, elp, point_names,
+                 nasion=None, lpa=None, rpa=None):
+        self.hsp = hsp
+        self.hpi = hpi
+        self.elp = elp
+        self.point_names = point_names
+
+        self.nasion = nasion
+        self.lpa = lpa
+        self.rpa = rpa
+
+    def __repr__(self):
+        s = '<DigMontage | %d Dig Points, %d HPI points: %s ...>'
+        s %= self.kind, len(self.ch_names), ', '.join(self.point_names[:3])
+        return s
+
+    def plot(self, scale_factor=1.5, show_names=False):
+        """Plot EEG sensor montage
+
+        Parameters
+        ----------
+        scale_factor : float
+            Determines the size of the points. Defaults to 1.5
+        show_names : bool
+            Whether to show the channel names. Defaults to False
+
+        Returns
+        -------
+        fig : Instance of matplotlib.figure.Figure
+            The figure object.
+        """
+        from ..viz import plot_montage
+        return plot_montage(self, scale_factor=scale_factor,
+                            show_names=show_names)
+
+
+def read_dig_montage(hsp=None, hpi=None, elp=None, point_names=None,
+                     transform=True):
+    """Read montage from a file
+
+    Note: Digitized points will be transformed to head-based coordinate system.
+
+    Parameters
+    ----------
+    hsp : None | str | array, shape (n_points, 3)
+        If str, this corresponds to the filename of the headshape points.
+        If numpy.array, this corresponds to an array of positions of the
+        channels in 3d.
+    hpi : None | str | numpy.array, shape (n_channels, 3)
+        If str, this corresponds to the filename of hpi points. If numpy.array,
+        this corresponds to an array hpi points. These points are in
+        device space.
+    elp : None | str | numpy.array
+        If str, this corresponds to the filename of hpi points. If numpy.array,
+        this corresponds to an array hpi points. These points are in
+        head space.
+    point_names : None | list
+        If list, this corresponds to a list of point names. This must be
+        specified if elp is defined.
+
+    Returns
+    -------
+    montage : instance of DigMontage
+        The digitizer montage.
+    """
+    if isinstance(hsp, str):
+        hsp = _read_dig_points(hsp)
+    if isinstance(hpi, str):
+        ext = op.splitext(hpi)[-1]
+        if ext == '.txt':
+            hpi = _read_dig_points(hpi)
+        elif ext in ('.sqd', '.mrk'):
+            hpi = read_mrk(hpi)
+        else:
+            raise TypeError('HPI file is not supported.')
+    if isinstance(elp, str):
+        elp_points = _read_dig_points(elp)
+    if len(elp) != len(point_names):
```

Comment:
These checks are not comprehensive. Think about possible all cases. For example, this will fail if `elp is None` or `point_names is None`. Also `elp` could be a string at this position in the code, which almost certainly isn't the comparison you want.